pkgs/by-name/sh/sharkey/package.nix

```diff
@@ -0,0 +1,184 @@
+      installPhase = ''
+        export HOME=$(mktemp -d)
+
+        pnpm config set store-dir $out
+        pnpm install --frozen-lockfile --ignore-script
+
+        rm -rf $out/v3/tmp
+        for f in $(find $out -name "*.json"); do
+          sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+          jq --sort-keys . $f | sponge $f
+        done
+      '';
+
+      dontBuild = true;
+      dontFixup = true;
+      outputHashMode = "recursive";
+      outputHash = "sha256-m+ue2KnAppgJtVQIfcgQ7MEvMePlsqpBquPDP25StUY=";
+    };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    nodePackages.pnpm
+    nodePackages.nodejs
+    makeWrapper
+    python3
+    pkg-config
+  ];
+
+  buildInputs = [
+    glib
+    vips
+  ];
+
+  patches = [
+    #./disable_update_checking.patch
+  ];
+
+  ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    export STORE_PATH=$(mktemp -d)
+    export NODE_OPTIONS = "--max_old_space_size=4096"
+
+    cp -Tr "$pnpmDeps" "$STORE_PATH"
+    chmod -R +w "$STORE_PATH"
+
+    pnpm config set store-dir "$STORE_PATH"
+    pnpm install --offline --frozen-lockfile --ignore-script
+    patchShebangs node_modules/{*,.*}
+  '';
+
+  postBuild = ''
+    pnpm build --reporter=ndjson
+  '';
+
+  # this is consistent with other nixpkgs electron packages and upstream, as far as I am aware
+  installPhase =
+    let
+      # this is mainly required for venmic
+      libPath = lib.makeLibraryPath ([
+        libpulseaudio
+        libnotify
+        pipewire
+        stdenv.cc.cc.lib
+      ] ++ lib.optional withTTS speechd);
+    in
+    ''
+      runHook preInstall
+
+      mkdir -p $out/data
+      cp dist/linux-*unpacked/resources/app.asar $out/opt/Vesktop/resources
+
+      mkdir -p $out/data/packages/client
+      ln -s /var/lib/misskey $out/data/files
+      ln -s /run/misskey $out/data/.config
+      cp -r locales node_modules built $out/data
+      cp -r packages/backend $out/data/packages/backend
+      cp -r packages/client/assets $out/data/packages/client/assets
+
+      # https://gist.github.com/MikaelFangel/2c36f7fd07ca50fac5a3255fa1992d1a
+
+      makeWrapper ${nodePackages.pnpm}/bin/pnpm $out/bin/sharkey \
+        --run $out/data
+        --prefix LD_LIBRARY_PATH : ${libPath} \
+        --prefix NODE_ENV : production
+        --add-flags start \
+        ${lib.optionalString withTTS "--add-flags \"--enable-speech-dispatcher\""} \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+
+      runHook postInstall
+    '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "vesktop";
+      desktopName = "Vesktop";
+      exec = "vesktop %U";
+      icon = "vesktop";
+      startupWMClass = "Vesktop";
+      genericName = "Internet Messenger";
+      keywords = [ "discord" "vencord" "electron" "chat" ];
+      categories = [ "Network" "InstantMessaging" "Chat" ];
+    })
+  ];
```

Comment:
I assume this also does not belong here ^^

Reply:
yep ^^

It's already removed @ my version locally :sweat_smile: